avogadro/rendering/geometryvisitor.h

```diff
@@ -41,7 +47,11 @@ class GeometryVisitor : public Visitor
   void visit(SphereGeometry&) override;
   void visit(AmbientOcclusionSphereGeometry&) override;
   void visit(CurveGeometry&) override;
+#ifdef _3DCONNEXION
+  void visit(CylinderGeometry&) override;
```

Comment:
If you want to replace this `visit()` method, you should do that (and whatever else you want to do in the implementation rather than an `ifdef` in the header.